Rough-draft of DD4EDU on Direct Deposit v2

erikphansen opened the pull request:
## Description
Still much to do, but this gets the Direct Deposit for EDU feature on the new Direct Deposit section of the Profile.

- [x] confirm that this has no effect on Direct Deposit v1 (when the `ch33_dd_profile` feature flag is off) **confirmed locally that user 1 sees their DD4CNP just fine and the new endpoint is not called when the feature flag is off**
- [ ] make the "update complete" alert to show and hide properly
- [ ] only show a single "set up 2FA" CTA, rather than one for each DD component
- [ ] since both new API endpoints need to be updated to provide the functionality we need, write Cypress tests that use a mock API to confirm things work

## Testing done
Local + new tests for updated selector functionality.

## Screenshots


## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

erikphansen committed:
Rename BankInfo components to match their file names

erikphansen committed:
make DD4EDU selectors less fragile

erikphansen committed:
Conditionally get the DD4EDU bank info

if the feature flag is turned on and the user is 2FA, call the new DD4EDU endpoint

erikphansen committed:
rough draft of Direct Deposit for EDU on the Direct Deposit page

erikphansen committed:
403 errors from GET DD4EDU endpoint will be treated as no-data state

We will show the user copy indicating that they do not appear to be eligible for EDU benefits

erikphansen committed:
remove extra vertical padding

erikphansen committed:
update support phone number

sanlouise commented on src/applications/personalization/profile/components/direct-deposit/BankInfoCNP.jsx:
```diff
@@ -160,7 +160,9 @@ export const DirectDepositCNP = ({
         className={classes.editButton}
-        aria-label={'Edit your direct deposit bank information'}
+        aria-label={
+          'Edit your direct deposit for disability compensation and pension benefits bank information'
```
Did you mean to also update this copy in v1? Looks good to me though since it's more specific 👍

erikphansen replied:
I did because why not? :-)

sanlouise commented on src/applications/personalization/profile/tests/selectors.unit.spec.js:
```diff
@@ -340,9 +340,13 @@ describe('profile selectors', () => {
 
   describe('eduDirectDepositLoadError', () => {
-    it('returns the error if it exists', () => {
+    it('returns any non-403 errors that exist', () => {
```
Thank you for adding these tests! 💯

erikphansen replied:
I don't know if I need additional comments somewhere to explain _why_ I added this special handling.

sanlouise replied:
I think your comment in the code `If the error is a 403 error, we will treat it like a no-data state, not an error` is sufficient!

erikphansen committed:
add link href

erikphansen committed:
fix broken unit test

erikphansen committed:
Add "not eligible for DD4CNP" content

We need this now because we will show the DD section if the user has DD4EDU set
up, even if they are not getting CNP benefits.
